velox/connectors/hive/storage_adapters/gcs/GCSFileSystem.cpp

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/hive/storage_adapters/gcs/GCSFileSystem.h"
+#include "velox/common/file/File.h"
+#include "velox/connectors/hive/storage_adapters/gcs/GCSUtil.h"
+#include "velox/core/Context.h"
+
+#include <fmt/format.h>
+#include <glog/logging.h>
+#include <memory>
+#include <stdexcept>
+
+#include <google/cloud/storage/client.h>
+
+namespace facebook::velox {
+namespace filesystems {
```

Comment:
`S3ReadFile` is not part of the `filesystems` namespace in S3. Can we follow the same here for `GCSReadFile` and `GCSWriteFile` to be consistent?